### PR TITLE
CustomCheckBox radio states

### PR DESCRIPTION
**Pull request by 84634E1A607A**
Updated radio button textures and added customCheckBox radio logic.

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: assets/textures/gui/bevel/radioA.png.import [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+[params]
+
+compress/mode=0
+compress/lossy_quality=0.7
+compress/hdr_mode=0
+compress/bptc_ldr=0
+compress/normal_map=0
+flags/repeat=0
+flags/filter=true
+flags/mipmaps=false
```
Comment (Kasterisk) [inline assets/textures/gui/bevel/radioA.png.import:24]: Mipmaps must be turned on.

File: src/gui_common/CustomCheckBox.cs [hunk 2]
```diff
@@ -27,6 +34,20 @@ private enum CheckState
...
+    public new ButtonGroup? Group
+    {
+        get => base.Group;
+        set
+        {
+            base.Group = value;
+
+            radio = Group != null;
+        }
+    }
```
Comment (Kasterisk) [inline src/gui_common/CustomCheckBox.cs:49]: Wouldn't it be more simpler to just check whether it's a radio button by doing something like?
```csharp
public bool Radio => Group != null;
```

**Review by Kasterisk — CHANGES_REQUESTED** [review]

File: src/gui_common/CustomCheckBox.cs (hunk 2 above)
Comment (84634E1A607A) [reply src/gui_common/CustomCheckBox.cs:49]: Makes sense

File: src/gui_common/CustomCheckBox.cs [hunk 3]
```diff
@@ -27,6 +33,8 @@ private enum CheckState
         Disabled,
     }
 
+    private bool Radio => Group != null;
```
Comment (Kasterisk) [inline src/gui_common/CustomCheckBox.cs:36]: Should be public, can be useful in other places.

**Review by Kasterisk — CHANGES_REQUESTED** [review]

**Review by Kasterisk — APPROVED** [review]

**Review by hhyyrylainen — APPROVED** [review]
I guess this isn't used anywhere yet so this can't be playtested yet. Code changes look fine.

**Comment by 84634E1A607A** [comment]
I had a quick test by adding several on the main scene. It looked fine. (But I didn't test radio box being disabled)

**Pull request merged**